mubench.reviewsite/src/Controller/FindingsUploader.php

```diff
@@ -24,45 +24,30 @@ public function processData($exp, $run)
         $this->logger->info("Data for : " . $detector->name);
         $project = $run->{'project'};
         $version = $run->{'version'};
-        $runtime = $run->{'runtime'};
-        $result = $run->{'result'};
-        $number_of_findings = $run->{'number_of_findings'};
         $potential_hits = $run->{'potential_hits'};
-
-        $this->createOrUpdateDetectorStatsTable($detector, $run);
+        $this->createOrUpdateTable($detector->getStatsTableName(), $run, array($this, 'createStatsTable'));
         $this->deleteAndStoreStats($detector, $exp, $project, $version, $run);
         if ($potential_hits) {
-            $this->createOrUpdateFindingsTable($detector, $potential_hits);
+            $this->createOrUpdateTable($detector->getTableName(), $potential_hits, array($this, 'createFindingsTable'));
             $this->storeFindings($exp, $detector, $project, $version, $potential_hits);
         }
     }
 
-    private function createOrUpdateDetectorStatsTable(Detector $detector, $run)
-    {
-        $propertyColumns = $this->getPropertyColumnNames($detector->getStatsTableName());
-        if (count($propertyColumns) == 0) {
-            $this->createStatsTable($detector);
-            $propertyColumns = ['exp', 'project', 'version'];
+    private function createOrUpdateTable($table_name, $object, $createFunc){
+        $existing_columns = $this->getPropertyColumnNames($table_name);
+        if (count($existing_columns) == 0) {
+            $existing_columns = $createFunc($table_name);
         }
-        $this->logger->info("Add columns to stats table" . $detector);
-        $this->addColumnsToTable($detector->getStatsTableName(), $propertyColumns, [$run]);
-    }
-
-    private function createOrUpdateFindingsTable(Detector $detector, $findings)
-    {
-        $propertyColumns = $this->getPropertyColumnNames($detector->getTableName());
-        if (count($propertyColumns) == 0) {
-            $this->createFindingsTable($detector);
-            $propertyColumns = ['exp', 'project', 'version', 'misuse', 'rank'];
+        $this->logger->info("Add columns to " . $table_name);
+        if(is_array($object)){
+            $propertyToColumnNameMapping = $this->getPropertyToColumnNameMapping($object);
+            $propertyToColumnNameMapping = $this->removeDisruptiveFindingsColumns($propertyToColumnNameMapping);
+        }else{
+            $propertyToColumnNameMapping = $this->getColumnNamesFromProperties($object);
+            $propertyToColumnNameMapping = $this->removeDisruptiveStatsColumns($propertyToColumnNameMapping);
```

Comment:
How about you pass the `$propertyColumnMapping` as an argument into this function, instead of `$object`? Because on the call site you know what this anyways and you don't need the `$object` for anything else in here anyways, right?

By the way, I really hate that you keep calling variables `$object` or similar names. That's not a name, it's a type. You can do better.